diff --git a/karchive/src/kar.cpp b/karchive/src/kar.cpp
--- a/karchive/src/kar.cpp
+++ b/karchive/src/kar.cpp
@@ -101,7 +101,7 @@ bool KAr::openArchive(QIODevice::OpenMode mode)
     if (mode == QIODevice::WriteOnly) {
         return true;
     }
     if (mode != QIODevice::ReadOnly && mode != QIODevice::ReadWrite) {
-        setErrorString(tr("Unsupported mode %1").arg(mode));
+        setErrorString(tr("Unsupported mode %1").arg(static_cast<int>(mode)));
         return false;
     }

diff --git a/karchive/src/karchive.cpp b/karchive/src/karchive.cpp
--- a/karchive/src/karchive.cpp
+++ b/karchive/src/karchive.cpp
@@ -168,4 +168,4 @@ bool KArchive::open(QIODevice::OpenMode mode)
     if (!d->dev->isOpen() && !d->dev->open(mode)) {
-        setErrorString(tr("Could not set device mode to %1").arg(mode));
+        setErrorString(tr("Could not set device mode to %1").arg(static_cast<int>(mode)));
         return false;
     }
@@ -215,4 +215,4 @@ bool KArchive::createDevice(QIODevice::OpenMode mode)
     default:
-        setErrorString(tr("Unsupported mode %1").arg(d->mode));
+        setErrorString(tr("Unsupported mode %1").arg(static_cast<int>(d->mode)));
         return false;
     }

diff --git a/karchive/src/krcc.cpp b/karchive/src/krcc.cpp
--- a/karchive/src/krcc.cpp
+++ b/karchive/src/krcc.cpp
@@ -123,7 +123,7 @@ bool KRcc::openArchive(QIODevice::OpenMode mode)
     if (mode == QIODevice::WriteOnly) {
         return true;
     }
     if (mode != QIODevice::ReadOnly && mode != QIODevice::ReadWrite) {
-        setErrorString(tr("Unsupported mode %1").arg(mode));
+        setErrorString(tr("Unsupported mode %1").arg(static_cast<int>(mode)));
         return false;
     }
